core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableSchemaBuilder.java

```diff
@@ -108,21 +115,36 @@ public void setSortColumns(List<ColumnSchema> sortColumns) {
   }
 
   public ColumnSchema addColumn(StructField field, boolean isSortColumn) {
+    return addColumn(field, null, isSortColumn, false);
+  }
+
+  private ColumnSchema addColumn(StructField field, String parentName, boolean isSortColumn,
+      boolean isComplexChild) {
     Objects.requireNonNull(field);
     checkRepeatColumnName(field);
     ColumnSchema newColumn = new ColumnSchema();
-    newColumn.setColumnName(field.getFieldName());
+    if (parentName != null) {
+      newColumn.setColumnName(parentName + "." + field.getFieldName());
+    } else {
+      newColumn.setColumnName(field.getFieldName());
+    }
     newColumn.setDataType(field.getDataType());
     if (isSortColumn ||
         field.getDataType() == DataTypes.STRING ||
```

Comment:
Instead of adding so many conditions in if add if(!(field.getDataType()==DataTypes.Double || field.getDataType()==DataTypes.BigDecimal)). Dimension column can be long, short, int, currently it is not handled

Reply:
In case if i have to place a negative condition then we have to add 
Non Dictionary 
-------------------
Boolean
short
int
float
long
double
Null
Byte


Dictionary 
-------------
string
array
struct
timestamp
date
char


AS dictionary check is smaller keeping it as it is.